### PR TITLE
Use MADR for Architecture Decisions

### PR DESCRIPTION
**Pull request by markwallace-microsoft**
### Motivation and Context
Please help reviewers and future users, providing the following information:
  1. Why is this change required? Proposal for how architectural decisions get documented and reviewed.
  2. What problem does it solve? Provides transparency in the architectural decision making and review process.
  3. What scenario does it contribute to? Will help to keep the language ports on the SK in synch.
  4. If it fixes an open issue, please link to the issue here.

### Description
This PR includes:

- README explaining how we will use MADR
- Two templates which are the starting point for new decision records
- One sample decision record which documents the proposal to use MADR

### Contribution Checklist
- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: docs/decisions/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+For more information [see](https://adr.github.io/madr/)
+
+## How are we using ADR's to track technical decisions?
+
+1. Copy docs/decisions/adr-template.md to docs/decisions/NNNN-title-with-dashes.md, where NNNN indicates the next number in sequence.
+    1. Check for existing PR's to make sure you use the correct sequence number.
+    2. There is also a short form template docs/decisions/adr-short-template.md
+2. Edit NNNN-title-with-dashes.md.
+    1. Status must initially be `proposed`
+    2. List of `deciders` must include the aliases of the people who will sign off on the decision.
```
Comment (adrianwyatt) [inline docs/decisions/README.md:13]: How do you feel about adding a comment to limit the list of deciders/approvers to a minimal set (e.g. 1-2)?

File: docs/decisions/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    3. The relevant EM and `dluc` must be listed as deciders or informed of all decisions.
```
Comment (adrianwyatt) [inline docs/decisions/README.md:14]: This is not necessarily true for all designs.

File: docs/decisions/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Markdown Any Decision Records
```
Comment (adrianwyatt) [inline docs/decisions/README.md:1]: Could we add explicit role definitions (i.e., decider, consultant, informed)?

**Review by adrianwyatt — CHANGES_REQUESTED** [review]

File: docs/decisions/README.md (hunk 2 above)
Comment (awharrison-28) [reply docs/decisions/README.md:14]: perhaps:

```suggestion
    3. The relevant EM and architect must be listed as deciders or informed of all decisions.
```

File: docs/decisions/README.md (hunk 3 above)
Comment (markwallace-microsoft) [reply docs/decisions/README.md:1]: The template files contains this description

File: docs/decisions/README.md (hunk 2 above)
Comment (markwallace-microsoft) [reply docs/decisions/README.md:14]: Done

File: docs/decisions/README.md (hunk 1 above)
Comment (markwallace-microsoft) [reply docs/decisions/README.md]: Done

**Review by lemillermicrosoft — APPROVED** [review]
Thanks again for the suggestion! I like this.

**Comment by lemillermicrosoft** [comment]
Thoughts on putting them in `decisions/` or `semantic-kernel/decisions/` or `knowledge-base/decisions/` instead of `docs/decisions`?

File: docs/decisions/0001-madr-architecture-decisions.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
+---
+# These are optional elements. Feel free to remove any of them.
+status: proposed
+date: 2023-05-29
+deciders: dluc,shawncal,hathind
```
Comment (lemillermicrosoft) [inline docs/decisions/0001-madr-architecture-decisions.md:5]:
```suggestion
deciders: dluc,shawncal,hathind,alliscode
```

**Review by shawncal — APPROVED** [review]

**Pull request merged**